backend/clubs/views.py

```diff
@@ -5354,6 +5354,52 @@ def transfer(self, request, *args, **kwargs):
     def get_queryset(self):
         return Ticket.objects.filter(owner=self.request.user.id)
 
+    def _give_tickets(self, user, order_info, cart, reconciliation_id):
+        """
+        Helper function that gives user/buyer their held tickets
+        and archives the transaction data
+        """
+
+        # At this point, we have validated that the payment was authorized
+        # Give the tickets to the user
+        tickets = cart.tickets.select_for_update().filter(holder=user)
+
+        # Archive transaction data for historical purposes.
+        # We're explicitly using the response data over what's in self.request.user
+        transaction_records = []
+
+        for ticket in tickets:
+            transaction_records.append(
+                TicketTransactionRecord(
+                    ticket=ticket,
+                    reconciliation_id=str(reconciliation_id),
+                    total_amount=float(order_info["amountDetails"]["totalAmount"]),
+                    buyer_first_name=order_info["billTo"]["firstName"],
+                    buyer_last_name=order_info["billTo"]["lastName"],
+                    # TODO: investigate why phone numbers don't show in test API
+                    buyer_phone=order_info["billTo"].get("phoneNumber", None),
+                    buyer_email=order_info["billTo"]["email"],
+                )
+            )
+
+        tickets.update(owner=self.request.user, holder=None)
+        TicketTransactionRecord.objects.bulk_create(transaction_records)
+        cart.tickets.clear()
+        for ticket in tickets:
+            ticket.send_confirmation_email()
+
+        Ticket.objects.update_holds()
+
+        cart.checkout_context = None
+        cart.save()
+
+    def _place_hold_on_tickets(self, tickets):
```

Comment:
Should probably add tests for this, like the tests we have for `_calculate_cart_total`.